ansi/parser_decode_test.go

```diff
@@ -0,0 +1,337 @@
+package ansi
+
+import (
+	"testing"
+)
+
+func TestDecodeSequence(t *testing.T) {
+	type expectedSequence struct {
+		seq   []byte
+		n     int
+		width int
+	}
+	cases := []struct {
+		name     string
+		input    []byte
+		expected []expectedSequence
+	}{
+		{
+			name:     "single byte",
+			input:    []byte{0x1b},
+			expected: []expectedSequence{{seq: []byte{0x1b}, n: 1}},
+		},
+		{
+			name:  "single byte 2",
+			input: []byte{0x00},
+			expected: []expectedSequence{
+				{seq: []byte{0x00}, n: 1},
+			},
+		},
+		{
+			name:  "ascii printable",
+			input: []byte("a"),
+			expected: []expectedSequence{
+				{seq: []byte{'a'}, n: 1, width: 1},
+			},
+		},
+		{
+			name:  "ascii space",
+			input: []byte(" "),
+			expected: []expectedSequence{
+				{seq: []byte{' '}, n: 1, width: 1},
+			},
+		},
+		{
+			name:  "ascii del",
+			input: []byte{DEL},
+			expected: []expectedSequence{
+				{seq: []byte{DEL}, n: 1},
+			},
+		},
+		{
+			name:  "del in the middle of utf8 string",
+			input: []byte{'a', DEL, 'b'},
+			expected: []expectedSequence{
+				{seq: []byte{'a'}, n: 1, width: 1},
+				{seq: []byte{DEL}, n: 1},
+				{seq: []byte{'b'}, n: 1, width: 1},
+			},
+		},
+		{
+			name:  "del in the middle of dcs",
+			input: []byte("\x1bP1;2+xa\x7fb\x1b\\"),
+			expected: []expectedSequence{
+				{seq: []byte("\x1bP1;2+xa\x7fb\x1b\\"), n: 12},
+			},
+		},
+		{
+			name:  "st in the middle of dcs",
+			input: []byte("\x1bP1;2+xa\x9cb\x1b\\"),
+			expected: []expectedSequence{
+				{seq: []byte("\x1bP1;2+xa\x9c"), n: 9},
+				{seq: []byte{'b'}, n: 1, width: 1},
+				{seq: []byte("\x1b\\"), n: 2},
+			},
+		},
+		{
+			name:     "csi",
```

Comment:
```suggestion
			name:     "CSI",
```

But I have no idea, what  is a CSI